src/_ert/forward_model_runner/fm_dispatch.py

```diff
@@ -1,20 +1,196 @@
+import argparse
+import contextlib
+import json
+import logging
 import os
 import signal
 import sys
+import time
+from collections.abc import Generator, Iterable
+from datetime import datetime
 
-from _ert.forward_model_runner.cli import main as job_runner_main
+from _ert.forward_model_runner import reporting
+from _ert.forward_model_runner.reporting.message import (
+    Finish,
+    Message,
+    ProcessTreeStatus,
+)
+from _ert.forward_model_runner.runner import ForwardModelRunner
+
+FORWARD_MODEL_DESCRIPTION_FILE = (
+    "jobs.json"  # This is wrongly named, but must be kept for backwards compatibility
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _setup_reporters(
+    is_interactive_run,
+    ens_id,
+    dispatch_url,
+    ee_token=None,
+    experiment_id=None,
+) -> list[reporting.Reporter]:
+    reporters: list[reporting.Reporter] = []
+    if is_interactive_run:
+        reporters.append(reporting.Interactive())
+    elif ens_id and experiment_id is None:
+        reporters.append(reporting.File())
+        reporters.append(reporting.Event(evaluator_url=dispatch_url, token=ee_token))
+    else:
+        reporters.append(reporting.File())
+    return reporters
+
+
+def _setup_logging(directory: str = "logs"):
+    fm_runner_logger = logging.getLogger("_ert.forward_model_runner")
+    memory_csv_logger = logging.getLogger("_ert.forward_model_memory_profiler")
+
+    os.makedirs(directory, exist_ok=True)
+
+    formatter = logging.Formatter(
+        "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+    )
+
+    filename = f"job-runner-log-{datetime.now().strftime('%Y-%m-%dT%H%M')}.txt"
+    csv_filename = f"memory-profile-{datetime.now().strftime('%Y-%m-%dT%H%M')}.csv"
+
+    handler = logging.FileHandler(filename=directory + "/" + filename)
+    handler.setFormatter(formatter)
+    handler.setLevel(logging.DEBUG)
+
+    csv_handler = logging.FileHandler(filename=directory + "/" + csv_filename)
+    csv_handler.setFormatter(logging.Formatter("%(message)s"))
+    memory_csv_logger.addHandler(csv_handler)
+    memory_csv_logger.setLevel(logging.INFO)
+    # Write the CSV header to the file:
+    memory_csv_logger.info(ProcessTreeStatus().csv_header())
+
+    fm_runner_logger.addHandler(handler)
+    fm_runner_logger.setLevel(logging.DEBUG)
+
+
+NFS_FILE_RETRY_TIME = 30
```

Comment:
Maybe just `FILE_RETRY_TIME` ? I guess this race condition could happen on other file systems also. Could maybe have a small comment.